aspnetcore/signalr/cpp-client.md

```diff
@@ -0,0 +1,103 @@
+---
+title: ASP.NET Core SignalR C++ client
+author: bradygaster
+description: Information about the ASP.NET Core SignalR C++ client
+monikerRange: '>= aspnetcore-3.0'
+ms.author: bradyg
+ms.custom: mvc
+ms.date: 02/26/2019
+uid: signalr/cpp-client
+---
+
+# ASP.NET Core SignalR C++ client
+
+The ASP.NET Core SignalR C++ client library lets you communicate with SignalR hubs from native applications.
+
+## Consume the SignalR C++ client library
+
+TODO
+
+## Connect to a hub
+
+```c++
+signalr::hub_connection connection("http://localhost:5000/default", signalr::trace_level::all);
+connection.start().get();
+```
+
+## Handle lost connection
+
+Register a handler for the disconnected event to respond to a lost connection. For example, you might want to automate reconnection.
+
+The disconnected event can be registered by calling `set_disconnected` and passing in a function:
+
+```c++
+connection.set_disconnected([]()
+{
+    // Do your close logic.
+});
+```
+
+## Call hub methods from client
+
+`connection.send` calls methods on the hub and doesn't expect or wait for a response.
+
+```c++
+web::json::value args{};
+args[0] = web::json::value::string("some text");
+connection.send("Echo", args).get();
+```
+
+`connection.invoke` calls methods on the hub and waits for a response.
+
+```c++
+web::json::value args{};
+args[0] = web::json::value::string("some text");
+connection.invoke("Echo", args)
+    .then([](const web::json::value& value)
+    {
+        ucout << value.serialize() << std::endl;
+    }).get();
+```
+
+Pass the hub method name and any arguments defined in the hub method to `invoke` or `send`.
+
+## Call client methods from hub
+
+Define methods the hub calls using `connection.on` after creating the connection, but before starting it. The `web::json::value` parameter will be an array of values sent from the server.
```

Comment:
For "but before starting it" you could note that you could miss messages otherwise